include/odp/api/spec/atomic.h

```diff
@@ -372,6 +372,30 @@ int odp_atomic_cas_u64(odp_atomic_u64_t *atom, uint64_t *old_val,
  */
 uint64_t odp_atomic_xchg_u64(odp_atomic_u64_t *atom, uint64_t new_val);
 
+
+/*
+ * 128-bit operations in RELAXED memory ordering
+ * --------------------------------------------
+ */
+
+/**
+ * Compare and swap atomic odp_u128_t variable
```

Comment:
Application needs an API defined way to read/write odp_u128_t data. The 128bit type would be mostly used to store e.g. two 64bit pointers (or array of smaller ints). It's less likely that calculations would be done directly with the 128b integer.

Options would be an union:
typedef union {
  __int128 u128;
  uint64_t u64[2];
  uint32_t u32[4];
...
} odp_u128_t;

... or accessor functions: odp_u128_from_u64(odp_u128_t *u128, uint64_t u64[]), odp_u128_to_u64(odp_u128_t *u128, uint64_t u64[]), ...

Reply:
I have added a union of different integer types. Please check.

One open question:
Currently, I have used __int128 data type. However, it is supported
in compiler version <GCC 4.6.4> and up. Whether support for compilers
older than <GCC 4.6.4> is required? The same question is applicable for
other compilers also. Currently, __int128 is being used in ODP internal code
and it has passed all the build configurations supported by Travis CI.
It means, is it enough to support GCC 4.6.4 and up for now ?